test/test_package_loading.py

```diff
@@ -0,0 +1,50 @@
+import os
+from pathlib import Path
+import time
+
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+import threading
+
+import pytest
+
+
+@pytest.fixture
+def static_web_server():
```

Comment:
We may want to standarize this with the webserver started in `tests/conftest.py`. In particular, if I understand correctly that one uses multiprocessing on some free port. Maybe moving both under some pytest fixture in `conftest.py` might make sense.

Reply:
I thought about using a fixture, but in this case, I truly wanted it to be a single webserver for *all* tests, so I didn't see much value.  

I've always preferred using multiprocessing for this because then the GIL isn't in contention when communicating between the webserver and the test process -- though in this case all the tests are in selenium (and thus a separate process anyway), so maybe it doesn't matter.

Maybe the thing to do is add a fixture to `conftest.py` that just returns the url and port to the singleton webserver:

```
@pytest.fixture
def web_server():
    yield '127.0.0.1', PORT
```

Then we wouldn't need this one.

Reply:
>  in this case, I truly wanted it to be a single webserver for all tests, so I didn't see much value.

I think a `pytest.fixture(scope='session')` would do just that?

Agreed that since we have free CPU cores, we might make use of them without worrying about the GIL.

> Maybe the thing to do is add a fixture to conftest.py that just returns the url and port to the singleton webserver:

Here, initially wanted to create a separate server with a different base URL / PORT to be sure that this is working as expected. You are right that re-using a single server might be sufficient though.